Penteract/Assets/Scripts/RatRobot.cpp

```diff
@@ -0,0 +1,152 @@
+#include "RatRobot.h"
+
+#include "GameplaySystems.h"
+#include "GameController.h"
+
+#include "Math/MathFunc.h"
+
+EXPOSE_MEMBERS(RatRobot) {
+	MEMBER(MemberType::GAME_OBJECT_UID, playerId),
+		MEMBER(MemberType::FLOAT, fleeRange),
+		MEMBER(MemberType::FLOAT, rotationSmoothness)
+};
+
+GENERATE_BODY_IMPL(RatRobot);
+
+void RatRobot::Start() {
+	player = GameplaySystems::GetGameObject(playerId);
+	if (!player) player = GameplaySystems::GetGameObject("Player");
```

Comment:
This might be dangerous when the UID hasn't been set properly because if it's incorrectly set, it definetly shouldn't work. If an occurrence of the Player exists in the Scene but hasn't been linked to this, we cannot be sure if the Rats should behave like this.